RecoTracker/MkFitCore/src/KalmanUtilsMPlex.cc

```diff
@@ -1103,23 +1240,598 @@ namespace mkfit {
+	gxu(n,0,0) += plPnt(n,0,0);
+	gxu(n,0,1) += plPnt(n,0,1);
+	gxu(n,0,2) += plPnt(n,0,2);
+      }
+      MPlexHV gpu;
+      RotateVectorOnPlaneTransp(rot, lpu, gpu);
+
+      for (int n = 0; n < N_proc; ++n) {//fixme vectorize,use cached pt?
+	outPar(n, 0, 0) = gxu.At(n, 0, 0);
+	outPar(n, 0, 1) = gxu.At(n, 0, 1);
+	outPar(n, 0, 2) = gxu.At(n, 0, 2);
+	const float pt = std::sqrt(gpu.At(n, 0, 0)*gpu.At(n, 0, 0) + gpu.At(n, 0, 1)*gpu.At(n, 0, 1));
+	outPar(n, 0, 3) = 1.f/pt;
+	outPar(n, 0, 4) = getPhi(gpu.At(n, 0, 0), gpu.At(n, 0, 1));
+	outPar(n, 0, 5) = getTheta(pt, gpu.At(n, 0, 2));
+      }
+
+      //now we need the jacobian to convert from curvilinear to CCS
+      // code from TrackState::jacobianCurvilinearToCCS
+      MPlex65  jacCurv2CCS(0.f); // fixme: cache sin and cosine, vectorize
+      for (int n = 0; n < N_proc; ++n) {
+	jacCurv2CCS(n, 0, 3) = -std::sin(outPar(n, 4, 0));
+	jacCurv2CCS(n, 0, 4) = -std::cos(outPar(n, 5, 0)) * std::cos(outPar(n, 4, 0));
+	jacCurv2CCS(n, 1, 3) = std::cos(outPar(n, 4, 0));
+	jacCurv2CCS(n, 1, 4) = -std::cos(outPar(n, 5, 0)) * std::sin(outPar(n, 4, 0));
+	jacCurv2CCS(n, 2, 4) = std::sin(outPar(n, 5, 0));
+	jacCurv2CCS(n, 3, 0) = inChg(n, 0, 0) / std::sin(outPar(n, 5, 0));
+	jacCurv2CCS(n, 3, 1) = outPar(n, 3, 0) * std::cos(outPar(n, 5, 0)) / std::sin(outPar(n, 5, 0));
+	jacCurv2CCS(n, 4, 2) = 1.f;
+	jacCurv2CCS(n, 5, 1) = -1.f;
+      }
+
+      //now we need the jacobian from local to curv
+      // code from TrackingTools/AnalyticalJacobians/src/JacobianLocalToCurvilinear.cc
+      MPlexHV tnl;
+      for (int n = 0; n < N_proc; ++n) {//fixme vectorize
+	tnl(n,0,0) = lpu(n,0,0)*std::max(std::abs(lp_upd(n,0,0)),1.e-9f);
+	tnl(n,0,1) = lpu(n,0,1)*std::max(std::abs(lp_upd(n,0,0)),1.e-9f);
+	tnl(n,0,2) = lpu(n,0,2)*std::max(std::abs(lp_upd(n,0,0)),1.e-9f);
+      }
+      MPlexHV tn;
+      RotateVectorOnPlaneTransp(rot, tnl, tn);
+      for (int n = 0; n < N_proc; ++n) {//fixme vectorize
+	vn(n,0,2) = std::max(1.e-30f,std::sqrt(tn(n,0,0)*tn(n,0,0)+tn(n,0,1)*tn(n,0,1)));
+	un(n,0,0) = -tn(n,0,1)/vn(n,0,2);
+	un(n,0,1) =  tn(n,0,0)/vn(n,0,2);
+	un(n,0,2) = 0.f;
+	vn(n,0,0) = -tn(n,0,2)*un(n,0,1);
+	vn(n,0,1) =  tn(n,0,2)*un(n,0,0);
+      }
+      MPlex55 jacLoc2Curv(0.f);
+      for (int n = 0; n < N_proc; ++n) {//fixme vectorize
+	// fixme? //(pf.use_param_b_field ? 0.01f * Const::sol * Config::bFieldFromZR(psPar(n, 2, 0), hipo(psPar(n, 0, 0), psPar(n, 1, 0))) : 0.01f * Const::sol * Config::Bfield);
+	const float bF = 0.01f * Const::sol * Config::Bfield;//fixme: cache?
+	const float qh2 = bF * lp_upd(n,0,0);
+	const float cosl1 = 1./vn(n,0,2);
+	const float uj = un(n,0,0)*rot(n,0,0) + un(n,0,1)*rot(n,0,1);
+	const float uk = un(n,0,0)*rot(n,1,0) + un(n,0,1)*rot(n,1,1);
+	const float vj = vn(n,0,0)*rot(n,0,0) + vn(n,0,1)*rot(n,0,1) + vn(n,0,2)*rot(n,0,2);
+	const float vk = vn(n,0,0)*rot(n,1,0) + vn(n,0,1)*rot(n,1,1) + vn(n,0,2)*rot(n,1,2);
+	const float cosz = vn(n,0,2)*qh2;
+	jacLoc2Curv(n,0,0) = 1.;
```

Comment:
```suggestion
	jacLoc2Curv(n,0,0) = 1.f;
```
although assignments should be OK

Reply:
https://github.com/trackreco/cmssw/blob/6f9a138a316977d892e404eb3e22b627e07ea1ce/RecoTracker/MkFitCore/src/Matriplex/Matriplex.h#L78-L85

Brute forcing the matrix to positive...